hoot-core/src/main/cpp/hoot/core/conflate/poi-polygon/PoiPolygonMatch.cpp

```diff
@@ -0,0 +1,497 @@
+//          {
+//            if (typeScore <= 0.64)
+//            {
+//              _class.setMatch();
+//            }
+//          }
+//          else
+//          {
+//            if (distance <= 0.000371)
+//            {
+//              _class.setMatch();
+//            }
+//          }
+//        }
+//      }
+//    }*/
+//  }
+//  catch (const geos::util::TopologyException& e)
+//  {
+//    //if (_badGeomCount <= ConfigOptions().getOgrLogLimit())
+//    //{
+//      LOG_WARN(
+//        "Feature(s) passed to PoiPolygonMatchCreator caused topology exception on conversion "
+//        "to a geometry: " << _poly->toString() << "\n" << _poi->toString() << "\n" << e.what());
+//      //_badGeomCount++;
+//    //}
+//    return;
+//  }
+
+//  LOG_VART(_class);
+//  LOG_TRACE("**************************");
+}
+
+void PoiPolygonMatch::_calculateMatch(const ElementId& eid1, const ElementId& eid2)
+{
+  _class.setMiss();
+
+  _categorizeElementsByGeometryType(eid1, eid2);
+
+  const unsigned int evidence = _calculateEvidence(_poi, _poly);
+  if (evidence >= MATCH_EVIDENCE_THRESHOLD)
+  {
+    _class.setMatch();
+  }
+  else if (evidence >= REVIEW_EVIDENCE_THRESHOLD)
+  {
+    _class.setReview();
+  }
+
+  LOG_VART(_class);
+  LOG_TRACE("**************************");
+}
+
+unsigned int PoiPolygonMatch::_getDistanceEvidence(ConstElementPtr poi, ConstElementPtr poly)
+{
+  _distance = PoiPolygonDistanceExtractor().extract(*_map, poi, poly);
+  if (_distance == -1.0)
+  {
+    _closeMatch = false;
+    return 0;
+  }
+
+  //search radius taken from PoiPolygonMatchCreator
+  PoiPolygonDistance distanceCalc(
+    _matchDistanceThreshold, _reviewDistanceThreshold, poly->getTags(),
+    poi->getCircularError() + ConfigOptions().getPoiPolygonReviewDistanceThreshold());
+  //type based match distance changes didn't have any positive effect experimentally; leaving it
+  //commented out here in case there is need for further examination
+//  _matchDistanceThreshold =
+//    max(
+//      distanceCalc.getMatchDistanceForType(_t1BestKvp),
+//      distanceCalc.getMatchDistanceForType(_t2BestKvp));
+  _reviewDistanceThreshold =
+    max(
+      distanceCalc.getReviewDistanceForType(_poi->getTags()),
+      distanceCalc.getReviewDistanceForType(_poly->getTags()));
+  //density based distance changes didn't have any positive effect experimentally; leaving it
+  //commented out here in case there is need for further examination
+  /*if (poi->getTags().get("station") != "light_rail" &&
+      poi->getTags().get("amenity") != "fuel")
+  {
+    distanceCalc.modifyMatchDistanceForPolyDensity(_matchDistanceThreshold);
+    distanceCalc.modifyReviewDistanceForPolyDensity(_reviewDistanceThreshold);
+  }*/
+
+  // calculate the 2 sigma for the distance between the two objects
+  const double sigma1 = poi->getCircularError() / 2.0;
+  const double sigma2 = poly->getCircularError() / 2.0;
+  const double ce = sqrt(sigma1 * sigma1 + sigma2 * sigma2) * 2;
```

Comment:
I'm pretty sure that 2*sqrt( (sigma1/2)*(sigma1/2) + (sigma2/2)*(sigma2/2)) == sqrt(sigma1*sigma1 + sigma2*sigma2)

Is there a reason to divide by 2, then multiply by 2 later?